ui/src/actions/holdingpen.ts

```diff
@@ -228,3 +234,53 @@ export function searchQueryUpdate(query: {
     dispatch(updateQuery(query));
   };
 }
+
+function resolvingAction(type: string) {
+  return {
+    type: HOLDINGPEN_RESOLVE_ACTION_REQUEST,
+    payload: { type },
+  };
+}
+
+function resolveActionSuccess() {
+  return {
+    type: HOLDINGPEN_RESOLVE_ACTION_SUCCESS,
+  };
+}
+
+function resolveActionError(errorPayload: { error: Error }) {
+  return {
+    type: HOLDINGPEN_RESOLVE_ACTION_ERROR,
+    payload: { ...errorPayload },
+  };
+}
+
+export function resolveAction(
+  id: string,
+  action: string,
+  payload: any
+): (dispatch: ActionCreator<Action>) => Promise<void> {
+  return async (dispatch) => {
+    dispatch(resolvingAction(action));
+    try {
+      await httpClient.post(
+        `${BACKOFFICE_API}/authors/${id}/${action}/`,
+        payload
+      );
+
+      dispatch(resolveActionSuccess());
+      notifyActionSuccess(action);
+      setTimeout(() => {
+        window?.location?.reload();
+      }, 3000);
```

Comment:
🤔

Reply:
This is in order to delay the reload of the page after successful decision. If the reload happens immediately, the user is unable to read the success notification and doesn't know that the decision was made successfully (because the status changes to 'running' only after 15-20 seconds). It will be deleted once we include this information (about possible decisions, as discussed yesterday) in the response.